src/libaktualizr/primary/aktualizr_fullostree_test.cc

```diff
@@ -0,0 +1,167 @@
+#include "httpfake.h"
+#include "logging/logging.h"
+#include "package_manager/ostreemanager.h"
+#include "package_manager/packagemanagerfactory.h"
+#include "primary/aktualizr.h"
+#include "primary/events.h"
+#include "primary/sotauptaneclient.h"
+#include "storage/sqlstorage.h"
+#include "test_utils.h"
+#include "uptane/tuf.h"
+#include "utilities/apiqueue.h"
+
+boost::filesystem::path aktualizr_repo_path;
+static std::string server = "http://127.0.0.1:";
+static std::string treehub_server = "http://127.0.0.1:";
+static boost::filesystem::path sysroot;
+
+static struct {
+  int serial;
+  std::string rev;
+} ostree_deployment;
+static std::string new_rev;
+
+#include <ostree.h>
+extern "C" OstreeDeployment *ostree_sysroot_get_booted_deployment_mock(OstreeSysroot *self) {
+  (void)self;
+  static GObjectUniquePtr<OstreeDeployment> dep;
+
+  dep.reset(ostree_deployment_new(0, "dummy-os", ostree_deployment.rev.c_str(), ostree_deployment.serial,
+                                  ostree_deployment.rev.c_str(), ostree_deployment.serial));
+  return dep.get();
+}
+
+extern "C" const char *ostree_deployment_get_csum(OstreeDeployment *self) {
+  (void)self;
+  return ostree_deployment.rev.c_str();
+}
+
+TEST(Aktualizr, FullOstreeUpdate) {
+  TemporaryDirectory temp_dir;
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, server);
+  conf.pacman.type = PackageManager::kOstree;
+  conf.pacman.sysroot = sysroot.string();
+  conf.pacman.ostree_server = treehub_server;
+  conf.pacman.os = "dummy-os";
+  conf.provision.device_id = "device_id";
+  conf.provision.ecu_registration_endpoint = server + "/director/ecus";
+  conf.provision.primary_ecu_serial = "CA:FE:A6:D2:84:9D";
+  conf.provision.primary_ecu_hardware_id = "primary_hw";
+  conf.provision.provision_path = "tests/test_data/cred.zip";
+  conf.tls.server = server;
+
+  LOG_INFO << "conf: " << conf;
+
+  {
+    Aktualizr aktualizr(conf);
+
+    aktualizr.Initialize();
+
+    result::UpdateCheck update_result = aktualizr.CheckUpdates().get();
+    ASSERT_EQ(update_result.status, result::UpdateStatus::kUpdatesAvailable);
+
+    result::Download download_result = aktualizr.Download(update_result.updates).get();
+    EXPECT_EQ(download_result.status, result::DownloadStatus::kSuccess);
+
+    result::Install install_result = aktualizr.Install(update_result.updates).get();
+    EXPECT_EQ(install_result.ecu_reports.size(), 1);
+    EXPECT_EQ(install_result.ecu_reports[0].install_res.result_code.num_code,
+              data::ResultCode::Numeric::kNeedCompletion);
+  }
+
+  // do "reboot" and finalize
+  ostree_deployment.serial = 1;
+  ostree_deployment.rev = new_rev;
+  boost::filesystem::remove(conf.bootloader.reboot_sentinel_dir / conf.bootloader.reboot_sentinel_name);
+
+  {
+    Aktualizr aktualizr(conf);
+
+    aktualizr.Initialize();
+
+    result::UpdateCheck update_result = aktualizr.CheckUpdates().get();
+    ASSERT_EQ(update_result.status, result::UpdateStatus::kNoUpdatesAvailable);
+  }
```

Comment:
I think it would be worth checking somehow that OSTree, the manifest/installation report, or the events sent to the server correctly reflect the expected installed version.

Reply:
I've added a check using OstreeManager's `getCurrent()`. Would it fit the bill?

Reply:
Yep, good enough, I think!